### DISCUSSION TITLE
Is there a way to allocate an instance of a class?

### DISCUSSION THREAD

**Discussion by NSExceptional** [Azoy/Echo#30] [Q&A] [answered]
I know there's some function like `swift_allocObject` or something that some other reflection libraries make use of, but I don't see anything like that in this repo

**Comment by NSExceptional** [comment, marked as answer]
Oh, I see it's actually in there already, but commented out! Why is it commented out?

**Reply by Azoy** [reply]
I took it out because I feared it might be too easy for people to get wrong, but looking back I could've written a wrapper that helps alleviate some of the things that could be wrong.

**Reply by NSExceptional** [reply]
Gotcha. Shall I make a PR?

**Reply by Azoy** [reply]
Had some changes last night that I forgot to push: https://github.com/Azoy/Echo/pull/31

**Reply by NSExceptional** [reply]
In terms of making it simple to use, it would be simplest if it only took the metadata and not also the size. Doesn't the size come from the metadata anyway?

**Reply by Azoy** [reply]
So I thought about that and it does make sense to only have a metadata argument because the other two do come directly from it, but there are use cases where one might want to tail allocate other elements that don't appear in the class's fields. First thing that comes to mind are KeyPaths, but I can imagine others can think of other use cases which is why I left the option open.

**Reply by NSExceptional** [reply]
Well in that case, would an `additionalBytes: Int = 0` argument make more sense instead? That way the primary use case requires the least effort. It could also be `sizeOverride: Int? = nil` or something, anything that allows the caller to just use the default size

Third idea: an enum like this

```swift
enum ObjectAllocationSize {
    case default
    case extraBytes(Int)
    case override(Int)
}

swift_allocObject(_ type: Metadata, options: ObjectAllocationSize = .default)
```
